src/state_machine.rs

```diff
@@ -111,8 +112,8 @@ impl State {
         }
     }
 
-    fn into_bootstrapped(self, proxy_peer_id: PeerId, proxy_public_id: PublicId) -> Self {
-        match self {
+    fn transition_to_final(self, proxy_peer_id: PeerId, proxy_public_id: PublicId) -> Self {
```

Comment:
I guess I agree with @afck on both points really.  Certainly `into` seems to be a standard Rust convention.